solidity/dashboard/src/services/tbtc-authorization.service.js

```diff
@@ -328,35 +336,50 @@ const fetchOperatorsOf = async (web3Context, yourAddress) => {
   for (let i = 0; i < managedGrantAddresses.length; ++i) {
     const managedGrantAddress = managedGrantAddresses[i]
     // operators of grantee (managedGrantAddress)
-    operators.push(
-      ...(await contractService.makeCall(
-        web3Context,
-        TOKEN_GRANT_CONTRACT_NAME,
-        "getGranteeOperators",
-        managedGrantAddress
-      ))
+    const operatorsOfManagedGrant = await contractService.makeCall(
+      web3Context,
+      TOKEN_GRANT_CONTRACT_NAME,
+      "getGranteeOperators",
+      managedGrantAddress
     )
+    for (let i = 0; i < operatorsOfManagedGrant.length; i++) {
+      operators.set(
+        web3Utils.toChecksumAddress(operatorsOfManagedGrant[i]),
+        false
+      )
+    }
   }
 
   // operators of authorizer
   const operatorsOfAuthorizer = await fetchOperatorsOfAuthorizer(web3Context)
-  operators.push(...operatorsOfAuthorizer)
+  for (let i = 0; i < operatorsOfAuthorizer.length; i++) {
+    operators.set(web3Utils.toChecksumAddress(operatorsOfAuthorizer[i]), false)
+  }
 
-  if (operators.length === 0) {
-    const ownerAddress = await contractService.makeCall(
-      web3Context,
-      TOKEN_STAKING_CONTRACT_NAME,
-      "ownerOf",
-      yourAddress
-    )
+  // operators of owner
+  const operatorsOfOwner = await contractService.makeCall(
+    web3Context,
+    TOKEN_STAKING_CONTRACT_NAME,
+    "operatorsOf",
+    yourAddress // as owner
+  )
+  for (let i = 0; i < operatorsOfOwner.length; i++) {
+    operators.set(web3Utils.toChecksumAddress(operatorsOfOwner[i]), true)
+  }
 
-    if (ownerAddress !== "0x0000000000000000000000000000000000000000") {
-      // owner of yourAddress
-      operators.push(yourAddress)
-    }
+  const ownerAddress = await contractService.makeCall(
+    web3Context,
+    TOKEN_STAKING_CONTRACT_NAME,
+    "ownerOf",
+    yourAddress
+  )
+
+  if (ownerAddress !== "0x0000000000000000000000000000000000000000") {
+    // yourAddress is an operator
+    operators.set(web3Utils.toChecksumAddress(yourAddress), true)
   }
 
-  return [...new Set(operators)]
+  return [...operators]
```

Comment:
Is there any reason why we return an array instead of Map?

Reply:
I guess it made sense while was drafting things. No reason why we should return an array. Will return a map to make further processing cleaner.